### ISSUE TITLE
Duplicate Entires in Big Picture Mode

### ISSUE THREAD

**Issue by DallonTG** [SteamGridDB/steam-rom-manager#229]
I can see multiple issues have been made regarding duplicate entries in the past, but I think this problem is different. 

Basically, whenever "Generate app list" is clicked, an entry is created for each game in Big Picture Mode - even if the exact same entry already exists. Note that duplicates are NOT created in the normal Steam mode, only Big Picture Mode.

I'm seeing this issue on Windows 10 with the latest release of Steam and the Steam Rom Manager.

**Comment by cbartondock** [comment]
I'm not able to reproduce this behavior. Did you maybe upgrade from a very old version of SRM that had added games? Can you try going into srm settings and hitting remove all added entries (warning: will remove all of your added games and reset artwork choices), verifying that all the stuff srm added to steam is gone, and then try again to add using SRM?

**Comment by DallonTG** [comment]
I only first installed SRM a couple of months ago. I have actually tried to remove all entries using the SRM settings before, but the option is greyed out and cannot be clicked. Steam is closed, I am an admin, and I have Windows User Account Control disabled.

**Comment by cbartondock** [comment]
That's very strange. That means SRM doesn't think you have added any items. Did you perchance delete a file in `C:\Program Files (x86)\Steam\userdata\[your steam id]\config` called `addedItemsV2.json`? If not, is that file there where it should be given that you say you have added games using SRM?

**Comment by DallonTG** [comment]
I haven't deleted it. The file is present and looks like this:

```
[
	"17733674215778811904_161033445263737354",
	"11586889838342701056_161033445263737354",
	"10368963840502136832_161033445263737354",
	"12854954631561740288_161033445263737354",
	"13278905583425028096_161033445263737354",
	"12041128109303398400_161033445263737354",
	"9826792165764759552_161033096081880387",
	"12569378691227844608_161033096081880387",
	"15705356376731549696_161033154837326515",
	"16231527176577482752_161033096081880387",
	"14422724755251527680_161033096081880387",
	"11510275734975283200_161033096081880387",
	"9432663564134383616_161033096081880387",
	"15640186076285370368_161033096081880387",
	"13444034590486298624_161033096081880387",
	"14417098734640824320_161033096081880387",
	"12719226187470077952_161033096081880387",
	"9828054344983904256_161033154837326515",
	"16128342498551529472_161033154837326515",
	"16373433617591828480_161033154837326515",
	"16332378098223808512_161033154837326515",
	"11383552802216214528_161033154837326515",
	"16828100869886050304_161033445263737354",
	"11304051196630663168_161033445263737354",
	"16708650893012107264_161033445263737354"
]
```

**Comment by cbartondock** [comment]
You could always try temporarily moving shortcuts.vdf that's in the same directory to a backup location, restarting steam to let it generate a new one, and trying again with SRM (this will wipe your shortcuts). It's very strange though, I've never seen that behavior and I can't think how it's possible that the behavior would be different for big picture vs non pig picture

**Comment by jamierajewski** [comment]
I can confirm the same issue. Not sure how it got duplicated, but I can also confirm that the "Remove All Entries" option is greyed out for me too. Tried opening/closing the app, regenerating my app list, deleting the config files and starting again, but it wont work.

**Comment by n8jadams** [comment]
I noticed this happening when I use two different computers... for some reason Steam ROM Manager games are being "synced" with the Steam account and added to the libraries of both computers (which is terrible because it's very unlikely someone is retaining the same file structure and the same roms on two machines)

**Comment by n8jadams** [comment]
I'm having a similar issue.

Windows 10. Standalone version. Every time I open Steam ROM Manager under "Preview" I only see "Generate app list", even though I already ran it.

Then when I run "Generate app list" and then "Save app list" it creates things correctly, but with duplicates for every entry under "Uncategorized." I'd be more than happy to jump on and pair with someone to replicate this bug and diagnose it.

**Comment by cbartondock** [comment]
I am guessing that somehow addedItemsV2.json is getting deleted for you guys... so SRM doesn't think it has added anything and it just re-adds everything. Did you totally delete your steam folders or something?

**Comment by cbartondock** [comment]
This issue is exactly what would occur if `C:\Program Files (x86)\Steam\userdata\XXXXXXXX\config\addedItemsV2.json` somehow got moved or deleted, or perhaps its permissions got changed so SRM can no longer read it? This is very difficult for me to debug as I have never had the issue, nor have I steps to reproduce it.

**Comment by cbartondock** [comment]
Closing this issue as it should be fixed in 2.3.16 (after doing remove all added entries).
